src/input_xml.F90

```diff
@@ -3823,6 +3910,46 @@ subroutine read_tallies_xml()
              // trim(to_str(t % id)) // ".")
       end if
 
+      ! Check for a tally derivative.
+      if (check_for_node(node_tal, "derivative")) then
+        ! Temporarily store the derivative id.
+        call get_node_value(node_tal, "derivative", t % deriv)
+
+        ! Find the derivative with the given id, and store it's index.
+        do j = 1, size(tally_derivs)
+          if (tally_derivs(j) % id == t % deriv) then
+            t % deriv = j
+            ! Only analog or collision estimators are supported for differential
+            ! tallies.
+            if (t % estimator == ESTIMATOR_TRACKLENGTH) &
+                 t % estimator = ESTIMATOR_COLLISION
+            exit
```

Comment:
`exit` is a rather confusingly named intrinsic in Fortran.  It means "exit this do loop" not "exit this program".  It's like the `break` statement in Python.  So this `exit` is simply to break out of the do loop that is searching for a derivative with a particular id.